Core/clim-core/table-formatting.lisp

```diff
@@ -294,31 +332,55 @@ skips intervening non-table output record structures."))
-      (setf (stream-cursor-position stream)
-            (values cursor-old-x cursor-old-y))
-      (with-output-recording-options (stream :record nil :draw nil)
-        (adjust-table-cells table stream)
-        (when multiple-columns (adjust-multiple-columns table stream))
-        (tree-recompute-extent table))
-      (replay table stream)
-      (if move-cursor
-          (setf (stream-cursor-position stream)
-                (values (bounding-rectangle-max-x table)
-                        (bounding-rectangle-max-y table)))
-          (setf (stream-cursor-position stream)
-                (values cursor-old-x cursor-old-y))))))
+  (apply #'invoke-with-new-output-record stream
+         (lambda (stream table)
+           (multiple-value-bind (cursor-old-x cursor-old-y)
+               (stream-cursor-position stream)
+             (with-output-recording-options (stream :record t :draw nil)
+               (funcall continuation stream)
+               (force-output stream))
+             (setf (stream-cursor-position stream)
+                   (values cursor-old-x cursor-old-y))
+             (with-output-recording-options (stream :record nil :draw nil)
+               (adjust-table-cells table stream)
+               (when multiple-columns (adjust-multiple-columns table stream))
+               (tree-recompute-extent table))
+             (replay table stream)
+             (setf (stream-cursor-position stream)
+                   (if move-cursor
+                       (values (bounding-rectangle-max-x table)
+                               (bounding-rectangle-max-y table))
+                       (values cursor-old-x cursor-old-y)))))
+         record-type
+         :x-spacing x-spacing
+         :y-spacing y-spacing
+         :multiple-columns multiple-columns
+         :multiple-columns-x-spacing multiple-columns-x-spacing
+         :equalize-column-widths equalize-column-widths
+         (alexandria:remove-from-plist
+          initargs :record-type :x-spacing :y-spacing :multiple-columns
+                   :multiple-columns-x-spacing :equalize-column-widths)))
+
+(defmacro formatting-table ((&optional (stream t)
+                             &rest args
+                             &key (record-type ''standard-table-output-record)
+                                  (move-cursor t)
+                                  . #.`(,@+standard-table-output-record-initargs+
+                                        &allow-other-keys))
+                            &body body)
+  (declare (ignore move-cursor . #.+standard-table-output-record-initargs+))
+  ;; If RECORD-TYPE is STANDARD-TABLE-OUTPUT-RECORD, make sure that
+  ;; all keyword arguments with constant keys correspond to known
+  ;; initargs.
+  (maybe-check-constant-keyword-arguments
+   record-type 'standard-table-output-record
+   args '(:move-cursor . #.(map 'list #'alexandria:make-keyword
+                            +standard-table-output-record-initargs+)))
```

Comment:
use keywords in the list so you don't have to map at read time but simply use the constant.